File: backend/app/schemas/revenue_analytics.py
from typing import List, Optional
from pydantic import BaseModel

class FunnelStageMetric(BaseModel):
    stage_key: str
    stage_label: str
    count: int
    value_inr: float
    conversion_from_prev_pct: float
    dropoff_count: int
    dropoff_pct: float

class ChannelAttributionMetric(BaseModel):
    source: str
    inquiries: int
    qualified: int
    site_visits: int
    bookings: int
    deal_value_inr: float
    commission_inr: float
    conversion_rate: float
    quality_score: int
    grade: str # "A+" | "A" | "B" | "C" | "D"

class RevenueAnalyticsResponse(BaseModel):
    time_period: str
    total_inquiries: int
    total_visits: int
    total_bookings: int
    total_closed_revenue: float
    total_commission: float
    overall_conversion_rate: float
    avg_cycle_days: float
    bottleneck_stage: str
    bottleneck_insight: str
    funnel_stages: List[FunnelStageMetric]
    channels: List[ChannelAttributionMetric]

File: frontend/src/types/revenueAnalytics.ts
export interface FunnelStageMetric {
  stage_key: string;
  stage_label: string;
  count: number;
  value_inr: number;
  conversion_from_prev_pct: number;
  dropoff_count: number;
  dropoff_pct: number;
}

export interface ChannelAttributionMetric {
  source: string;
  inquiries: number;
  qualified: number;
  site_visits: number;
  bookings: number;
  deal_value_inr: number;
  commission_inr: number;
  conversion_rate: number;
  quality_score: number;
  grade: 'A+' | 'A' | 'B' | 'C' | 'D' | string;
}

export interface RevenueAnalyticsResponse {
  time_period: string;
  total_inquiries: number;
  total_visits: number;
  total_bookings: number;
  total_closed_revenue: number;
  total_commission: number;
  overall_conversion_rate: number;
  avg_cycle_days: number;
  bottleneck_stage: string;
  bottleneck_insight: string;
  funnel_stages: FunnelStageMetric[];
  channels: ChannelAttributionMetric[];
}
